mlir/lib/Dialect/MIOpen/Transforms/CopyOpt.cpp

```diff
@@ -59,6 +59,7 @@ template <typename T> struct MICORewritePattern : public OpRewritePattern<T> {
     // 1. Capture allocation->copy pattern
     Operation *writer = nullptr;
     memref::CopyOp reader = nullptr;
+    memref::CollapseShapeOp csop = nullptr;
```

Comment:
This should be generalized for all subsequent collapse/expand ops. Not required for this commit but please file a ticket with testcase for multiple reshapes in various combinations.

Reply:
https://github.com/ROCmSoftwarePlatform/llvm-project-private/issues/644 filed.